react-app/src/components/GridPage.tsx

```diff
@@ -0,0 +1,170 @@
+import { CSSProperties, useState } from "react";
+import styles from "../styles/GridPage.module.css";
+import { colors } from "../static/GridElements";
+import GridConfigButton from "./UI/GridButton";
+import GridButtonWindow from "./GridButtonWindow";
+import { GridConfigSlider } from "./GridConfigSlider.tsx";
+import { subtractWithSaturation8bit } from "../utils.tsx";
+import { CodeListing } from "./CodeListing.tsx";
+import Grid from "./Grid.tsx";
+
+export default function GridPage() {
+  const numberOfElements = 20;
+
+  const [cssProps, setCssProps] = useState<CSSProperties>({
+    gridAutoFlow: "dense",
+    gridTemplateColumns: "100",
+    gridAutoRows: "100",
+    gap: "10",
+  });
+
+  const [selectedAspects, setSelectedAspects] = useState([
+    { id: 1, aspect: "aspect_1_to_2", selected: true },
+    { id: 2, aspect: "aspect_2_to_1", selected: true },
+    { id: 3, aspect: "aspect_1_to_1", selected: true },
+    { id: 4, aspect: "aspect_3_to_1", selected: true },
+    { id: 5, aspect: "aspect_1_to_3", selected: true },
+  ]);
+
+  const [elements, setElements] = useState(getGeneratedElements);
```

Comment:
I am not fan of this approach, I think the better way of handling it is the array of element data and constructing the component from that